### PR TITLE
Support for secret param in material

### PR DESCRIPTION
**Pull request by bdpiparva**
link to issue #6014

### CODE REVIEW

File: domain/src/main/java/com/thoughtworks/go/domain/materials/git/GitMaterialUpdater.java [hunk 1]
```diff
@@ -140,7 +140,7 @@ private BuildCommand cleanWorkingDir(String workDir) {
 
     private BuildCommand isRepoUrlChanged(String workDir) {
         return test("-neq",
-                material.getUrlArgument().forCommandline(),
+                material.getUrlArgument().rawUrl(),
```
Comment (maheshp) [inline domain/src/main/java/com/thoughtworks/go/domain/materials/git/GitMaterialUpdater.java]: This should be changed back to `forCommandline()`

File: domain/src/main/java/com/thoughtworks/go/domain/materials/git/GitMaterialUpdater.java [hunk 2]
```diff
@@ -159,7 +159,7 @@ private BuildCommand cmdClone(String workDir, int cloneDepth) {
         if (material.isShallowClone()) {
             cloneArgs.add(format("--depth=%s", String.valueOf(cloneDepth)));
         }
-        cloneArgs.add(material.getUrlArgument().forCommandline());
+        cloneArgs.add(material.getUrlArgument().rawUrl());
```
Comment (maheshp) [inline domain/src/main/java/com/thoughtworks/go/domain/materials/git/GitMaterialUpdater.java]: This should be changed back to forCommandline()

File: domain/src/main/java/com/thoughtworks/go/domain/materials/git/GitMaterialUpdater.java (hunk 1 above)
Comment (bdpiparva) [reply domain/src/main/java/com/thoughtworks/go/domain/materials/git/GitMaterialUpdater.java]: done

File: domain/src/main/java/com/thoughtworks/go/domain/materials/git/GitMaterialUpdater.java (hunk 2 above)
Comment (bdpiparva) [reply domain/src/main/java/com/thoughtworks/go/domain/materials/git/GitMaterialUpdater.java]: done

File: config/config-api/src/main/java/com/thoughtworks/go/util/command/UrlArgument.java [hunk 3]
```diff
@@ -35,26 +36,42 @@ public UrlArgument(String url) {
...
+    }
+
+    @Override
+    public String originalArgument() {
         return url;
     }
 
+    @Override
     public String forDisplay() {
+        if (hasSecretParams()) {
```
Comment (maheshp) [inline config/config-api/src/main/java/com/thoughtworks/go/util/command/UrlArgument.java]: We might have to change this, e.g `https://#{SECRET[id][key]}:password@foo.com/bar.git` how will this be handled?

**Pull request merged**